mapie/multi_label_classification.py

```diff
@@ -0,0 +1,738 @@
+
+        Returns
+        -------
+        ClassifierMixin
+            The estimator itself or a default multi-output
+            ``LogisticRegression`` instance.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not ``None``
+            and has no fit, predict, nor predict_proba methods.
+
+        NotFittedError
+            If the estimator is not fitted.
+        """
+        if estimator is None:
+            return MultiOutputClassifier(
+                LogisticRegression(multi_class="multinomial")
+            ).fit(X, y)
+
+        if isinstance(estimator, Pipeline):
+            est = estimator[-1]
+        else:
+            est = estimator
+        if (
+            not hasattr(est, "fit")
+            or not hasattr(est, "predict")
+            or not hasattr(est, "predict_proba")
+        ):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a classifier with fit,"
+                "predict, and predict_proba methods."
+            )
+        check_is_fitted(est)
+        return estimator
+
+    def _check_partial_fit_first_call(self) -> bool:
+        """Check that this is the first time partial_fit
+        or fit is called.
+
+        Returns
+        -------
+        bool
+            True if it is the first time, else False.
+        """
+        return not hasattr(self, "risks")
+
+    def _check_bound(self, bound: Optional[str]):
+        """Check the value on the bound.
+
+        Parameters
+        ----------
+        bound : Optional[str]
+            Bound defined in the predict.
+
+        Raises
+        ------
+        AttributeError
+            If bound in not in ["hoeffding", "bernstein", "wsr", None]
+        """
+        if bound not in ["hoeffding", "bernstein", "wsr", None]:
+            raise ValueError(
+                'bound must be in ["hoeffding", "bernstein", "wsr", None]'
+            )
+
+    def _transform_pred_proba(
+        self,
+        y_pred_proba: Union[Sequence[NDArray], NDArray]
+    ) -> NDArray:
+        """If the output of the predict_proba is a list of arrays (output of
+        the ``predict_proba`` of ``MultiOutputClassifier``) we transform it
+        into an array of shape (n_samples, n_classes)
+
+        Parameters
+        ----------
+        y_pred_proba : Union[List, NDArray]
+            Output of the multi-label classifier.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes)
+            Output of the model ready for risk computation.
+        """
+        if type(y_pred_proba) == np.ndarray:
+            y_pred_proba_array = y_pred_proba
+        else:
+            y_pred_proba_stacked = np.stack(
+                y_pred_proba,  # type: ignore
+                axis=0
+            )[:, :, 1]
+            y_pred_proba_array = np.moveaxis(y_pred_proba_stacked, 0, -1)
+
+        return np.expand_dims(y_pred_proba_array, axis=2)
```

Comment:
you do something anyway: should you tell it for developpers?

Reply:
I don't understand this comment

Reply:
that's because I need to do something different according to the type of y_pred_proba

Reply:
Yes indeed, but precise in the docstring that you change the data structure anyway (the docstring implicitely tells that you do something only if the output of the predict_proba is a list of arrays)

Reply:
done